Tests/scripts/secrets.py

```diff
@@ -0,0 +1,223 @@
+
+        # Open each file, read its contents in UTF-8 encoding to avoid unicode characters
+        with io.open('./' + file_path, mode="r", encoding="utf-8") as file:
+            file_contents = file.read()
+
+            # Add all context output paths keywords to whitelist temporary
+            temp_white_list = create_temp_white_list(yml_file_contents if yml_file_contents else file_contents)
+            secrets_white_list = secrets_white_list.union(temp_white_list)
+
+            # Search by lines after strings with high entropy as possibly suspicious
+            for line in file_contents.split('\n'):
+
+                # if detected disable-secrets comment, skip the line
+                if bool(re.findall(r'(disable-secrets-detection-start)', line)):
+                    skip_secrets = True
+                if bool(re.findall(r'(disable-secrets-detection-end)', line)):
+                    skip_secrets = False
+                if skip_secrets or bool(re.findall(r'(disable-secrets-detection)', line)):
+                    continue
+
+                # REGEX scanning for IOCs and false positive groups
+                potential_secrets, false_positives = regex_for_secrets(line)
+                for potential_secret in potential_secrets:
+                    if not any(white_list_string in potential_secret for white_list_string in secrets_white_list):
+                        regex_secrets.append(potential_secret)
+                # added false positives into white list array before testing the strings in line
+                secrets_white_list = secrets_white_list.union(false_positives)
+
+                # calculate entropy for each string in the file
+                for string_ in line.split():
+                    string_ = string_.strip("\"()[],'><:;\\")
+                    string_lower = string_.lower()
+                    # compare the lower case of the string against both generic whitelist & temp white list
+                    if not any(white_list_string in string_lower for white_list_string in secrets_white_list):
+                        entropy = calculate_shannon_entropy(string_)
+                        if entropy >= ENTROPY_THRESHOLD:
+                            high_entropy_strings.append(string_)
+
+        if high_entropy_strings or regex_secrets:
+            # uniquify identical matches between lists
+            file_secrets = list(set(high_entropy_strings + regex_secrets))
+            secrets_found[file_name] = file_secrets
+
+    return secrets_found
+
+
+def create_temp_white_list(file_contents):
+    temp_white_list = set([])
+    context_paths = re.findall(r'contextPath: (\S+\.+\S+)', file_contents)
+    for context_path in context_paths:
+        context_path = context_path.split('.')
+        context_path = [white_item.lower() for white_item in context_path]
+        temp_white_list = temp_white_list.union(context_path)
+
+    return temp_white_list
+
+
+def retrieve_related_yml(file_path_temp):
```

Comment:
https://github.com/demisto/content/pull/2911/files
Scripts/ParseEmailFiles/parse_email_files_test.py

Reply:
you haven't fixed this

Reply:
I actually see both ParseEmailFiles.py and ParseEmailFiles.yml,
do we have a better way to match them?
If not it is not critical since the worst thing that will happen is that i won't retrieve a temp white list from there, the script will keep working fine with possibly more false positives, that is all.